compiler/rustc_trait_selection/src/traits/select/candidate_assembly.rs

```diff
@@ -717,32 +716,35 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         };
 
         let obligation = traits::Obligation::new(
-            cause.clone(),
+            ObligationCause::dummy(),
```

Comment:
Uh, why losing cause?

Reply:
Doesn't really matter here because we're just checking. But yeah, can add it back.